src/App.js

```diff
@@ -1,5 +1,61 @@
+const { Console } = require("@woowacourse/mission-utils");
+const { GAME_MESSAGE, RESTART, END } = require("./util/Constant");
+const makeNumber = require("./util/MakeNumber");
+const isValidNum = require("./util/IsValidNum");
+const { makeComment, makeCount } = require("./util/MakeCount");
+
 class App {
-  play() {}
-}
+  play() {
+    Console.print(GAME_MESSAGE.START_MESSAGE);
+    this.gameStart();
+  }
+
+  gameStart() {
+    this.answer = makeNumber();
+    this.inputAnswer();
+  }
+  gameEnd() {
+    Console.print(GAME_MESSAGE.END_MESSAGE);
+    Console.readLine(GAME_MESSAGE.INTENTION_MESSAGE, (input) => {
+      this.askRestart(input);
+    });
+  }
+  askRestart(input) {
+    if (input === RESTART) return this.gameStart();
+    if (input === END) return Console.close();
+    return throwErr();
+  }
+
+  inputAnswer() {
+    Console.readLine(GAME_MESSAGE.INPUT_MESSAGE, (input) => {
+      this.userInput = Array.from(String(input), Number);
+      this.checkInput();
+    });
+  }
+  checkInput() {
+    if (!isValidNum(this.userInput)) {
+      this.throwErr();
+    }
+    this.makeOutput();
+  }
+  makeOutput() {
+    const { strike, ball } = makeCount(this.answer, this.userInput);
+    const comment = makeComment(strike, ball);
 
+    Console.print(comment);
+
+    if (strike !== 3) {
+      this.inputAnswer();
+    } else {
+      this.gameEnd();
+    }
+  }
```

Comment:
```suggestion
    if (strike === 3) {
      this.gameEnd();
      return;
    } 
    this.inputAnswer();
  }
```
이런 식으로 early return을 이용해서 else를 쓰지 않도록 하는 것은 어떨까요? 이번 3주차 부터는 else를 지양한다 라는 요구 사항이 생겨서 적용해보시면 좋을 것 같아요!